src/index.js

```diff
@@ -27,6 +65,42 @@ const products = [
 // 7. Target the button in index.html, attach an event listener to it. Every time we click
 // on the button, it will toggle the product list. Show => hide, hide => show
 
+const ulist = document.querySelector(".product-list");
+const btnToggle = document.querySelector(".toggle");
+var searchInput = document.querySelector(".search-input");
+const res = document.querySelector(".result")
+const form = document.querySelector(".product-form")
+btnToggle.addEventListener('click', function (){
+  products.forEach((element)=>{
+    const newLi = document.createElement('li')
+    newLi.innerText = element.name;
+    ulist.appendChild(newLi)
+  })
+  btnToggle.style.visibility = 'visible'? 'hidden': 'visible';
+})
+form.addEventListener('submit',function(e){
+  e.preventDefault()
+  res.textContent = "";
+  if(searchInput.value !== "") {
+    console.log(searchInput.value)
+    var found = products.find((product)=> product.name === searchInput.value);
```

Comment:
Not a mistake, but as potential improvement: In the gif example, partially completely search items also give results. For example, "pai" would show "paint", "ca" would show "car". Might be nice to implement.